### PR TITLE
feat: make uds cli first class

### PR DESCRIPTION
**Pull request by decleaver**
## Description

Sets architecture for deploy if not specified

## Related Issue

Relates to #195 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: src/pkg/bundle/deploy.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 	"golang.org/x/exp/slices"
 	"helm.sh/helm/v3/pkg/cli/values"
 	"helm.sh/helm/v3/pkg/getter"
+	// "oras.land/oras-go/v2/registry"
```
Comment (UncleGedd) [inline src/pkg/bundle/deploy.go:30]:
```suggestion
```

File: src/pkg/bundle/pull.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ func (b *Bundler) Pull() error {
 		return err
 	}
 
+	// Get validated source path
+	b.cfg.PullOpts.Source = getOciValidatedSource(b.cfg.PullOpts.Source)
```
Comment (UncleGedd) [inline src/pkg/bundle/pull.go:32]:
```suggestion
	b.cfg.PullOpts.Source = getOCIValidatedSource(b.cfg.PullOpts.Source)
```

File: src/pkg/bundle/remote.go [hunk 3]
```diff
@@ -224,3 +233,72 @@ func (op *ociProvider) PublishBundle(_ types.UDSBundle, _ *oci.OrasRemote) error
 	// todo: implement moving bundles from one registry to another
 	return fmt.Errorf("moving bundles in between remote registries not yet supported")
 }
+
+// Returns the validated source path based on the provided oci source path
+func getOciValidatedSource(source string) string {
```
Comment (UncleGedd) [inline src/pkg/bundle/remote.go:238]:
```suggestion
func getOCIValidatedSource(source string) string {
```

**Comment by UncleGedd** [comment]
Made a couple of nit suggestions but overall looks good!

File: src/pkg/bundle/deploy.go (hunk 1 above)
Comment (decleaver) [reply src/pkg/bundle/deploy.go]: updated

File: src/pkg/bundle/pull.go (hunk 2 above)
Comment (decleaver) [reply src/pkg/bundle/pull.go]: updated

File: src/pkg/bundle/remote.go (hunk 3 above)
Comment (decleaver) [reply src/pkg/bundle/remote.go]: updated

File: src/pkg/bundle/remote.go [hunk 4]
```diff
@@ -224,3 +233,72 @@ func (op *ociProvider) PublishBundle(_ types.UDSBundle, _ *oci.OrasRemote) error
...
+	arch := a[len(a)-1]
+	// confirm arch is valid
+	return slices.Contains(config.GetSupportedArchitectures(), arch)
+}
+
+// CheckOCISourcePath checks that provided oci source path is valid, and updates it if it's missing the full path
+func CheckOCISourcePath(source string) string{
+	validTarballPath := utils.IsValidTarballPath(source)
+	if !validTarballPath {
+		source = getOciValidatedSource(source)
```
[the remark this answers is not in the document]
Comment (decleaver) [reply src/pkg/bundle/remote.go]: updated

**Review by UncleGedd — APPROVED** [review]
ship it! 🚀

**Pull request merged**